### DISCUSSION TITLE
Measure scattering parameters on VNA simultaneously

### DISCUSSION THREAD

**Discussion by tvstefanski** [microsoft/Qcodes#3909] [General]
Hello,

I recently needed to measure all 4 scattering parameters (S11, S22, S12, and S21) to perform characterization of a filter. In theory, the VNA should allow for all four of these to be measured simultaneously, however it seems that in line 965 you are only able to use QCoDeS to perform one measurement per channel. I have temporarily implemented a loop over the scattering parameters, but I was wondering if there is a more streamlined way to do this?

Thanks in advance!

**Comment by jenshnielsen** [comment]
Hi, Is this the ZNB VNAs? 

The way is is set up in those drivers there is a channel for each parameter. These can either be added manually or the instrument can be setup to capture all s parameters. Since each parameter is a channel I think you should be able to do

vna.channels.trace()

to get a trace from all channels at the same time
or 

vna.channels[1:3].trace()

to select a channel 1 and 2 as an example


Note however that due to current limitations that does not work for parameters which are multiparameters such as `trace_mag_phase`
